### PR TITLE
Add multi-user support

### PR DESCRIPTION
**Pull request by stefanv**
This PR also improves the WebSocket setup.

To test this branch, you need to have https://github.com/python-social-auth/social-storage-peewee/pull/1 installed

TODO:

- [x] Move token to configuration file
- [x] Disable username (or use login/pass) for testing purposes

### CODE REVIEW

File: cesium_app/handlers/base.py [hunk 1]
```diff
@@ -48,19 +46,49 @@ def prepare(self):
...
+        return super(BaseHandler, self).prepare()
+
+    def get_current_user(self):
+        if not cfg['server']['multi_user']:
+            username = 'testuser@gmail.com'
+            try:
+                models.User.get(username=username)
+            except models.User.DoesNotExist:
+                models.User.create(username=username, email=username)
+            return 'testuser@gmail.com'
```
Comment (acrellin) [inline cesium_app/handlers/base.py]: `return username`? doubly hard-coded

File: static/scripts/actions.jsx [hunk 2]
```diff
@@ -571,7 +616,9 @@ export function fetchPredictions() {
       dispatch,
       FETCH_PREDICTIONS,
 
-      fetch('/predictions')
+      fetch('/predictions', {
+        credentials: 'same-origin'
```
Comment (acrellin) [inline static/scripts/actions.jsx:620]: Perhaps this should be the default param, which would clean these calls up a bit.

**Comment by acrellin** [comment]
Looking great so far, @stefanv! Nice work

**Comment by acrellin** [comment]
@stefanv that little addition gets all tests passing except one on my machine

**Comment by stefanv** [comment]
Woah, neat!  Thanks @acrellin

File: cesium_app/tests/conftest.py [hunk 3]
```diff
@@ -35,11 +38,23 @@ def close():
...
+    # Authenticate by clicking login button
+    driver.get('/')
+    try:
+        driver.implicitly_wait(1)
+        driver.find_element_by_xpath('//a[@href="login/google-oauth2"]').click()
+    except NoSuchElementException:
+        # Already logged in
+        pass
+    driver.implicitly_wait(1)
+    assert driver.find_element_by_xpath('//div[contains(text(), "testuser@gmail.com")]')
```
Comment (stefanv) [inline cesium_app/tests/conftest.py:76]: Great!

File: tools/install_python_deps.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+
+import subprocess
+import os
+from os.path import join as pjoin
+
+with open(pjoin(os.path.dirname(__file__), '../requirements.txt')) as f:
+    deps = f.readlines()
+
+for dep in deps:
```
Comment (bnaul) [inline tools/install_python_deps.py]: Combine as `for dep in open(pjoin(os.path.dirname(__file__), '../requirements.txt')):`?

File: cesium_app/config.py [hunk 5]
```diff
@@ -1,125 +1,51 @@
...
-def show_config():
-    """Print config settings to stdout (run on app start)."""
-    print()
-    print("=" * 78)
-    print("cesium configuration")
-
-    for key in ('paths', 'database', 'testing', 'docker'):
-        if key in cfg:
+        if ':' in key:
+            keys = key.split(':')
```
Comment (bnaul) [inline cesium_app/config.py]: No need for the `if` block; `'a'.split(':') == ['a']`.

**Review by bnaul — APPROVED** [review]

**Review by acrellin — APPROVED** [review]

**Pull request merged**